RATIONALE.md

```diff
@@ -186,6 +192,89 @@ extensible, meaning that third-party may define new effects and implement
 externally the associated effect handlers in order to customize Nickel for
 specific use-cases.
 
+## Why Nickel is not a DSL embedded in an existing language
+
+Using an existing language to embed configuration is a common approach. In this
+section, we'll try to answer the question why Nickel isn't a DSL embedded in,
+say, Haskell.
+
+Beside typechecking, the Nix language itself could also be the host of a
+configuration DSL. This is in fact exactly what Nixpkgs and NixOS modules are, a
+DSL embedded into the Nix language.
```

Comment:
This paragraph, while it makes perfect sense when responding to a question from Valentin, seems out of place here.

Reply:
Agreed, will get rid of it :+1: